adam/language_specific/chinese/chinese_language_generator.py

```diff
@@ -0,0 +1,477 @@
+import collections
+from itertools import chain
+from typing import Iterable, List, Mapping, MutableMapping, Optional, Tuple, Union, cast
+from attr import Factory, attrib, attrs
+from attr.validators import instance_of
+from immutablecollections import ImmutableSet, immutableset, immutablesetmultidict
+from more_itertools import first, only
+from networkx import DiGraph
+from adam.axes import FacingAddresseeAxis, GRAVITATIONAL_DOWN_TO_UP_AXIS
+from adam.language.dependency import (
+    DependencyRole,
+    DependencyTree,
+    DependencyTreeLinearizer,
+    DependencyTreeToken,
+    LinearizedDependencyTree,
+)
+from adam.language.dependency.universal_dependencies import (
+    PARTICLE,
+    ADJECTIVAL_MODIFIER,
+    ADPOSITION,
+    ADVERB,
+    ADVERBIAL_MODIFIER,
+    CASE_POSSESSIVE,
+    CASE_SPATIAL,
+    DETERMINER,
+    DETERMINER_ROLE,
+    INDIRECT_OBJECT,
+    NOMINAL_MODIFIER,
+    NOMINAL_MODIFIER_POSSESSIVE,
+    NOMINAL_SUBJECT,
+    NUMERAL,
+    NUMERIC_MODIFIER,
+    OBJECT,
+    OBLIQUE_NOMINAL,
+    PROPER_NOUN,
+    VERB,
+    IS_ATTRIBUTE,
+)
+from adam.language.language_generator import LanguageGenerator
+from adam.language.lexicon import LexiconEntry
+from adam.language.ontology_dictionary import OntologyLexicon
+from adam.language_specific.chinese.chinese_phase_1_lexicon import (
+    GAILA_PHASE_1_CHINESE_LEXICON,
+    ME,
+    YOU,
+)
+from adam.language_specific import (
+    FIRST_PERSON,
+    SECOND_PERSON,
+    ALLOWS_DITRANSITIVE,
+    MASS_NOUN,
+)
+from adam.language_specific.chinese.chinese_syntax import (
+    SIMPLE_CHINESE_DEPENDENCY_TREE_LINEARIZER,
+)
+from adam.ontology import IN_REGION, IS_ADDRESSEE, IS_SPEAKER, OntologyNode
+from adam.ontology.phase1_ontology import (
+    AGENT,
+    COLOR,
+    FALL,
+    GOAL,
+    GROUND,
+    HAS,
+    LEARNER,
+    PATIENT,
+    SIT,
+    THEME,
+    JUMP,
+)
+from adam.ontology.phase1_spatial_relations import (
+    EXTERIOR_BUT_IN_CONTACT,
+    GRAVITATIONAL_DOWN,
+    INTERIOR,
+    PROXIMAL,
+    Region,
+    TOWARD,
+    GRAVITATIONAL_UP,
+)
+from adam.random_utils import SequenceChooser
+from adam.relation import Relation
+from adam.situation import Action, SituationObject, SituationRegion
+from adam.situation.high_level_semantics_situation import HighLevelSemanticsSituation
+
+"""Simple rule-based approach for translating HighLevelSemanticSituations to 
+Chinese dependency trees. We currently only generate a single possible linearized
+tree for a semantic situation."""
+
+
+@attrs(frozen=True, slots=True)
+class SimpleRuleBasedChineseLanguageGenerator(
+    LanguageGenerator[HighLevelSemanticsSituation, LinearizedDependencyTree]
+):
+    # mapping from nodes in the concept ontology to Chinese words
```

Comment:
Put the docstring from above here. That way Python knows to attach it to the class.

Reply:
I'll use 📖 to mark the same problem elsewhere.